### PR TITLE
remove inactive users from all orgs except judges from JudgeTeams

### LINKED ISSUE #12940: Remove inactive users from all organizations
As a followup to #12303, we would like to remove a user from their organizations when they are marked inactive. This excludes removing judges from their `JudgeTeam`s as that will be handled in another ticket

## AC
- [ ] When a user is made inactive, they are removed from every organization they are a member or admin of, other than judges from `JudgeTeam`s

## Notes
As judge teams are the only organization that somewhat rely on there being a dedicated admin (at least until SCM epic is finished), there should be no ill effects of an organization that loses an admin.

### PR DESCRIPTION
**Pull request by yoomlam**
Resolves #12940 

### Description
Remove a user from their organizations when they are marked inactive, except do not remove judges from their JudgeTeam (they are marked inactive #12942).

Changes:
```diff
-    remove_user_from_auto_assign_orgs
+    remove_user_from_orgs
```

### Acceptance Criteria
- [x] Code compiles correctly and tests pass

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -288,6 +288,14 @@ def selectable_organizations
     orgs
   end
 
+  def judge?
+    if FeatureToggle.enabled?(:use_judge_team_role)
```
Comment (hschallhorn) [inline app/models/user.rb]: Even if we are not using judge team roles, we still want to check if there is a judge team before . we make a call to vacols. I'd remove this check.

File: app/models/user.rb [hunk 2]
```diff
@@ -352,13 +360,14 @@ def user_reactivation
...
+    remove_user_from_orgs
     JudgeTeam.for_judge(self)&.inactive!
   end
 
-  def remove_user_from_auto_assign_orgs
-    auto_assign_orgs = organizations.select(&:automatically_assign_to_member?)
-    auto_assign_orgs.each do |organization|
+  def remove_user_from_orgs
+    removal_orgs = organizations
+    removal_orgs = removal_orgs.reject { |org| org.is_a?(JudgeTeam) } if judge?
```
Comment (hschallhorn) [inline app/models/user.rb]: Here we're only checking if the user is a member of a judge team. Instead, we should reject the team that the user is the judge for. I'd add a check here to make sure `org.judge == self`

Comment (hschallhorn) [reply app/models/user.rb]: There are a lot of places in the code where we check `org.is_a?(JudgeTeam)` in a loop. We should offload this to the database. `removal_orgs.where.not(type: JudgeTeam.name)`. I'
ll file a ticket to clean up the rest of the code.

Comment (yoomlam) [reply app/models/user.rb]: In the new ticket, we should be careful of potential future subclasses of `JudgeTeam` that won't be found by `.where.not(type: JudgeTeam.name)`.

Comment (hschallhorn) [reply app/models/user.rb]: But also, that second comment I made was nullified by the first 🤷‍♂comment.

Comment (yoomlam) [reply app/models/user.rb]: The `.where.not(type: JudgeTeam.name)` also does not catch organizations with `type=nil` in tests that `create(:organization)`.

Comment (hschallhorn) [reply app/models/user.rb]: Hm, we should update factory bot to include types then. On that now.

Comment (hschallhorn) [reply app/models/user.rb]: #13177

**Comment by hschallhorn** [comment]
Let's user the new `judge?` function [here](https://github.com/department-of-veterans-affairs/caseflow/pull/13172/files#diff-4676c008b11a5480d73d4a6de01e45b9L278) as well.

Adding `before { allow(user).to receive(:judge?).and_return(false) }` [here](https://github.com/department-of-veterans-affairs/caseflow/pull/13172/files#diff-12b107c16792b9ecba685e51b51826f1R750) may solve the flake as well

File: app/models/user.rb [hunk 3]
```diff
@@ -362,9 +362,9 @@ def user_inactivation
...
   def remove_user_from_orgs
     removal_orgs = organizations
-    removal_orgs = removal_orgs.reject { |org| org.is_a?(JudgeTeam) && org.judge == self } if judge?
-    removal_orgs.each do |organization|
-      OrganizationsUser.remove_user_from_organization(self, organization)
+    my_judge_team = JudgeTeam.for_judge(self)
+    removal_orgs.each do |org|
+      OrganizationsUser.remove_user_from_organization(self, org) unless org == my_judge_team
     end
   end
```
Comment (yoomlam) [inline app/models/user.rb]: @hschallhorn Significantly changed the code after your review. What do you think about this current version?

Comment (hschallhorn) [reply app/models/user.rb]: 👌 I like it! This will probably error for an inactive judge team until #13169 is merged

**Pull request merged**